### PR TITLE
Fixed unicode char handling in Uris in Cosmos Http Client

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
Cosmos Http client incorrectly uses URI.toString instead of URI.toASCIIString when constructing the Netty Http request - resulting in incorrect handling of Unicode chars in a Uri on the wire. Using URI.toASCIIString instead fixes this.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java [hunk 1]
```diff
@@ -255,15 +255,15 @@ public void idWithUnicodeCharacters() {
...
+                HttpConstants.StatusCodes.OK,
+                HttpConstants.StatusCodes.OK,
+                HttpConstants.StatusCodes.NO_CONTENT),
             new TestScenarioExpectations(
                 "COMPUTE_GATEWAY",
                 HttpConstants.StatusCodes.CREATED,
-                HttpConstants.StatusCodes.BADREQUEST,// Bug in Compute Gateway - check with Dmitri when fix is available
-                HttpConstants.StatusCodes.BADREQUEST,
-                HttpConstants.StatusCodes.BADREQUEST),
+                HttpConstants.StatusCodes.OK,// Bug in Compute Gateway - check with Dmitri when fix is available
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java:264]: does the comment still true?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**